### PR TITLE
TR-4008 - test session restoration

### PR DESCRIPTION
**Pull request by poyuki**
# [TR-4008](https://oat-sa.atlassian.net/browse/TR-4008)

## Summary
Usage of test session state restoration functionality at delivery execution reactivation 

## Dependencies 
depends on https://github.com/oat-sa/extension-tao-testqti/pull/2258

## TAE
in process

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
         "oat-sa/extension-tao-group": ">=7.0.0",
         "oat-sa/extension-tao-item": ">=11.0.0",
         "oat-sa/extension-tao-test": ">=15.0.0",
-        "oat-sa/extension-tao-testqti": ">=44.13.0"
+        "oat-sa/extension-tao-testqti": " dev-feat/TR-4008/session_state_restoration"
```
Comment (kilatib) [inline composer.json:30]: should be commited?

**Review by wazelin — COMMENTED** [review]
Looks great!

Will approve as soon as the dependency's been released.

**Review by kilatib — APPROVED** [review]

File: helper/SessionStateHelper.php [hunk 2]
```diff
@@ -15,34 +15,74 @@
...
     }
+
+    private static function getTestSessionService(): TestSessionService
+    {
+        return ServiceManager::getServiceManager()->getContainer()->get(TestSessionService::SERVICE_ID);
+    }
+
+    private static function getTestSessionStateRestorationService(): TestSessionStateRestorationServiceInterface
+    {
+        return ServiceManager::getServiceManager()->getContainer()->get(TestSessionStateRestorationServiceInterface::class);
```
Comment (wazelin) [inline helper/SessionStateHelper.php:81]: This service seem to have been renamed.

**Review by wazelin — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

⚠️ just don't forget to update the dependency before merging ⚠️

**Pull request merged**